hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java

```diff
@@ -81,18 +81,16 @@ public void register(Statement statement, boolean cancelable) {
 	public void release(Statement statement) {
 		log.tracev( "Releasing statement [{0}]", statement );
 
-		// Keep this at DEBUG level, rather than warn.  Numerous connection pool implementations can return a
-		// proxy/wrapper around the JDBC Statement, causing excessive logging here.  See HHH-8210.
-		if ( log.isDebugEnabled() && !xref.containsKey( statement ) ) {
-			log.unregisteredStatement();
+		final Set<ResultSet> resultSets = xref.remove( statement );
+		if ( resultSets != null ) {
+			closeAll( resultSets );
 		}
 		else {
-			final Set<ResultSet> resultSets = xref.get( statement );
-			if ( resultSets != null ) {
-				closeAll( resultSets );
-			}
-			xref.remove( statement );
+			// Keep this at DEBUG level, rather than warn.  Numerous connection pool implementations can return a
+			// proxy/wrapper around the JDBC Statement, causing excessive logging here.  See HHH-8210.
+			log.unregisteredStatement();
 		}
+
```

Comment:
cannot be done the same for `public void release(ResultSet resultSet, Statement statement)` when `statement ! null`